hazelcast/protocol/builtin.py

```diff
@@ -496,3 +554,203 @@ def encode(buf, value, is_final=False):
     @staticmethod
     def decode(msg):
         return msg.next_frame().buf.decode("utf-8")
+
+
+class ListCNFixedSizeCodec(object):
+    _TYPE_NULL_ONLY = 1
+    _TYPE_NOT_NULL_ONLY = 2
+    _TYPE_MIXED = 3
+
+    _ITEMS_PER_BITMASK = 8
+
+    _HEADER_SIZE = BYTE_SIZE_IN_BYTES + INT_SIZE_IN_BYTES
+
+    @staticmethod
+    def decode(msg, item_size, decoder):
+        frame = msg.next_frame()
+        type = FixSizedTypesCodec.decode_byte(frame.buf, 0)
+        count = FixSizedTypesCodec.decode_int(frame.buf, 1)
+
+        if type == ListCNFixedSizeCodec._TYPE_NULL_ONLY:
+            return [None] * count
+        elif type == ListCNFixedSizeCodec._TYPE_NOT_NULL_ONLY:
+            header_size = ListCNFixedSizeCodec._HEADER_SIZE
+            return [decoder(frame.buf, header_size + i * item_size) for i in range(count)]
+        else:
+            response = []
```

Comment:
you can use fixed size list for better performance. see https://stackoverflow.com/a/10617221

```python
response = [None]*count
// some lines...
response[i] = decoder(frame.buf, position)
```